### DISCUSSION TITLE
Why no database?

### DISCUSSION THREAD

**Discussion by ebergerly** [LibreHardwareMonitor/LibreHardwareMonitor#1431] [General]
It seems like much of the coding work being done on this project is adding new hardware. And it looks like all of this data (sensor name, type, etc.) is actuallly hard coded, rather than in an external database?

Why?

Wouldn't it be a whole lot easier and require much less coding if it was in a SQLite or other simple database, or even a CSV file?? With a database you can also force standard naming conventions by giving the user a dropdown of sensor types, etc., to choose from. And you can do all kinds of filtering & searching very easily. And you can even give defaults to the fields based on what is known about the specific hardware. And it seems a simple C# winforms app to do data entry should be fairly simple to set up, and could even provide instructions on where to find the data.

**Comment by ebergerly** [comment]
And here's a quick winforms demo of the basic concepts. Just grab the existing hard coded enums with mfr, model name, etc., and convert to tables in a database. And it's got some dropdown combo boxes to quickly show what motherboard models are already entered (of course the combo box list depends on the manufacturer already selected), and if not, just hit the Add button and another form pops up to allow you to add the specifics (like Super I/O mfr, model, sensor data, etc.). This way even a non-coder could add a good portion of any new hardware, assuming they're at least referencing other existing apps like HWMonitor that might already have that hardware's data. 

Of course, this does give the coders one less thing to play with...

![DatabaseApp](https://github.com/user-attachments/assets/52bd7062-40d0-431a-bdf5-1548ac882f31)

**Comment by ebergerly** [comment]
So there's 580 lines of code in "Identification.cs" whose sole purpose is to make sure the mfr and model names are in the desired format?? 

"when name.StartsWith("Sony"), return Manufacturer.Sony"???

Wouldn't a dropdown that grabs desired names from a database, allowing the user to select a "correct" name, get rid of all that??

**Comment by Xcalibur37** [comment]
I think the main issue here is inherited technical debt. From other conversations I believe the maintainers like the idea - it would just take some time to migrate to that vs the way it was staged by the original author. That's my take on reading the history.

**Comment by Xcalibur37** [comment]
To add to this since I saw a comment that was since removed - sure, adding high level attributes isn't hard and could be put in a database. The real issue is the nuance between boards - especially when you are looking at Super I/O. Having spent a bit of time on this it's incredibly nuanced. Now does that mean "it's not possible?", No. It's harder than it looks. If you dive into the code for the specific chipsets you will see what I am talking about. I think it will take some extra planning to decouple.

I am currently working on this on my own fork which I plan to separate entirely from LibreHardwareMonitor in an attempt to refactor the entire code base. This is one area I plan to attack - adding configs without a new release (aside from an updated conf).
